### DISCUSSION TITLE
Add a per-slide timer to speaker notes?

### DISCUSSION THREAD

**Discussion by kquinsland** [marp-team/marp#529] [Ideas]
I really really really like the speaker notes view on marp! The clock/elapsed time is <3.
I have one request/idea that I think could make the speaker notes view that much more powerful...

Allow me to indicate the maximum amount of time that I want to spend on this slide so this can also be displayed in the speaker notes. I don't need a timer progress bar (although that would be _awesome_). A simple `00:05 / 00:30` display to indicate that I've been on this slide for 5 seconds so far and I didn't want to spend more than 30s would be adequate!

At present, I do something like this:

```md
<!--
[30s]

This is quick about me; spend at most 30s here; mostly to allow time for people to scan QR code on screen...

-->
```

Which does put a `[30s]` right at the top of my speaker notes. This has a few big down sides, though:

- I have to mentally keep track of the elapsed time when I entered this particular slide
- Then do the mental math to figure out what time I should be off this slide by

And ideally instantly and "in the background" while I continue to speak about the slide's content.

If there was a per-slide field like `timer` that I could configure like so:

```
---

<!-- _header: Current section: foobar -->

# Advantages of foobar

- blah
- blah

<!-- _footer: something snarky here :) -->
<!-- _timer: {max:30s, warn:20s} -->
<!-- 
If the audience was receptive to earlier interactive portion, ask $questions and link reply to the advantages.
...
-->

---
```

That would put an "elapsed time on $this slide" counter on the speaker notes view. After I've been on this slide for 5 seconds, the section of speaker notes would display `00:05/00:30` and at 20 seconds elapsed on this slide, the `00:20/00:30` would turn yellow (customizable via some css class, of course!) and at `00:30/00:30` the timer would be red as would any "overage": `01:10/00:30`  (if I spent 70 seconds on a slide that was only supposed to get 30...)

**Comment by yhatt** [comment]
Your suggestion would be able to implement with [Marp Core](https://github.com/marp-team/marp-core), with using [the custom local directive](https://marpit.marp.app/directives?id=custom-local-directive), the handmade [Marp plugins](https://marpit-api.marp.app/), and making the custom presentor tool.

Marp team is focusing on simple interface for slide writing experience, the presentation support for presentor is _intentionally_ minimum. From [the classic Marp app](https://github.com/yhatt/marp/issues?q=), we have learned that the scope creep will cause bankruptcy of the project. 😢 So probably we would not lead that.

We want to leave enhancing for presenters to third-party Marp tools or external tools for presenter integrated with Marp.
